src/main/java/com/nscharrenberg/um/multiagentsurveillance/headless/utils/GameOver.java

```diff
@@ -0,0 +1,108 @@
+package com.nscharrenberg.um.multiagentsurveillance.headless.utils;
+
+import com.nscharrenberg.um.multiagentsurveillance.headless.exceptions.BoardNotBuildException;
+import com.nscharrenberg.um.multiagentsurveillance.headless.exceptions.InvalidTileException;
+import com.nscharrenberg.um.multiagentsurveillance.headless.exceptions.ItemNotOnTileException;
+import com.nscharrenberg.um.multiagentsurveillance.headless.models.GameMode;
+import com.nscharrenberg.um.multiagentsurveillance.headless.Factory;
+import com.nscharrenberg.um.multiagentsurveillance.headless.contracts.repositories.IMapRepository;
+import com.nscharrenberg.um.multiagentsurveillance.headless.contracts.repositories.IPlayerRepository;
+import com.nscharrenberg.um.multiagentsurveillance.headless.models.Items.Item;
+import com.nscharrenberg.um.multiagentsurveillance.agents.shared.algorithms.distanceCalculator.ManhattanDistance;
+import com.nscharrenberg.um.multiagentsurveillance.headless.models.Map.Area;
+import com.nscharrenberg.um.multiagentsurveillance.headless.models.Map.Tile;
+import com.nscharrenberg.um.multiagentsurveillance.headless.models.Map.TileArea;
+import com.nscharrenberg.um.multiagentsurveillance.headless.models.Player.Guard;
+import com.nscharrenberg.um.multiagentsurveillance.headless.models.Player.Intruder;
+
+
+public class GameOver {
+
+    private IMapRepository mapRepository;
+    private IPlayerRepository playerRepository;
+
+    private ManhattanDistance manhattanDistance;
+
+    private static GameMode gameMode;
+
+    private static int caught = 0;
+    private static int escaped = 0;
+
+    public enum gameOver
+    {
+        WIN,
+        LOSE
+    }
+
+    public GameOver()
+    {
+        this.mapRepository = Factory.getMapRepository();
+        this.playerRepository = Factory.getPlayerRepository();
+    }
+
+    // Check whether the gameMode is Guard vs Intruder
+    public static boolean checkGameMode(){
+        return gameMode.getName().equals("Guard vs Intruder") || gameMode.getId() == 1;
+    }
+
+    // Check if all the intruders are gone
+    public static gameOver findIntruder(TileArea board, Tile intruderPos)

```

Comment:
The name and description of this function is not what the logic of this function is doing.